Pre-GRC Tuning

GreenTomato5 opened the pull request:
Adds CAN IDs taken from last years configs for each module in constants. These should be CHECKED PLEASE because the IDs may be mixed up and likley are. This is because we didn't properly name our modules last year when using YAGSL so "Front left" in configs wasn't actually front left. But the wheels should still move. I also added the NavX IO abstraction which currently uses the MXP port. This may need to be adjusted to use USB C but for now it should be fine.

GreenTomato5 committed:
Adds NAVX abstraction

GreenTomato5 committed:
Adds NAVX abstraction

GreenTomato5 committed:
CAN configs for each module

GreenTomato5 committed:
CAN configs for each module

GreenTomato5 committed:
Adds PID/FF controller values for REAL module

GreenTomato5 committed:
Adds PID/FF controller values for REAL module

Commit:
Apply Prettier format

Commit:
Apply Prettier format

GreenTomato5 committed:
Nah constants checkign is mad annoying ngl

GreenTomato5 committed:
Nah constants checkign is mad annoying ngl

Commit:
Apply Prettier format

Commit:
Apply Prettier format

PotmanNob commented on src/main/java/frc/robot/subsystems/drive/GyroIONavx2.java:
```diff
@@ -0,0 +1,44 @@
+package frc.robot.subsystems.drive;
+
+import com.kauailabs.navx.frc.AHRS;
```
Looks fine, but could you post the link to download the NavX library since I don't think i have it yet

GreenTomato5 replied:
Here: 
    https://dev.studica.com/releases/2024/NavX.json
    The vendor dep json thing should be in the branch so you don't have to re-install it

PotmanNob commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -269,26 +278,33 @@ public static final class Hybrid {
+				// TODO: Confirm module IDs are associated with correct module and tune module offsets if last years dont work
+
+				// Front Left Module
+				public static final int DRIVE0_ID = 8;
```
I don't have access to the robot until recruitment week, so I can't double check CAN ids. You should check again just in case or get someone else there to do it for you.

PotmanNob reviewed (approved):
Looks good, just double check the CAN ids just in case.

GreenTomato5 committed:
NavX was the problem, terrible code still :(

GreenTomato5 committed:
Oops

Commit:
Apply Prettier format

GreenTomato5 commented:
Problem is bc talons and sparks are using a different odo thread and the talons use blocking (so they wait for motor controller to report data and that adds latency so the data collected isnt aligned with the timestamps of the spark data collected) and the sparks use polling (just ask the sparks for data). Idk how to effectivley mix the two sampling types so the odo data works and it would probably take a lot of time so I'm gona switch it to the non advanced swerve template so we can get to testing important stuff faster

GreenTomato5 committed:
like probably 25% chance this works in real life

Commit:
Apply Prettier format

GreenTomato5 commented:
I lied, made a mixed thread so should maybe work

GreenTomato5 committed:
Womp womp

Commit:
Apply Prettier format

GreenTomato5 committed:
deletes log files, adds them to git ignore

GreenTomato5 committed:
Merge branch 'Fix-Drive-Configs' of https://github.com/FRC-7525/2024-Rewrite into Fix-Drive-Configs

GreenTomato5 committed:
Ohhh

Commit:
Apply Prettier format

GreenTomato5 committed:
terrible seizure robot

Commit:
Apply Prettier format

GreenTomato5 commented:
Todo (After sept 9, putting this here so we don't forget):
- Fix absolute encoder offsets, idk why they weren't offsetting
- Add back sys id and test it for drive controller values (once we get ground) 
- PID tune (and probably change the setpoints) of intake, amp bar, and shooter states
- Write climbers
- Automatic state transitions based on at setpoint, timers, and current sensing
- Add beam breaks or sumth

GreenTomato5 committed:
Fix format on sim/deploy

GreenTomato5 committed:
Delete unused code

GreenTomato5 committed:
Deletes more spotless stuff

GreenTomato5 committed:
:(

SSBASLD committed:
Fixed 3 issues

Autocommands redundant code
AppliedVoltage logging
Shooter using BangBangController

SSBASLD committed:
improved output logging so that it is distinct from inputs

SSBASLD committed:
readded input logging of module applied voltages

SSBASLD committed:
implemented suggested changes

SSBASLD committed:
Merge pull request #27 from SSBASLD/issues-fixes

Fixed 3 issues

Commit:
Apply Prettier format

SSBASLD committed:
added beam break functionality

SSBASLD committed:
changed automatically true to automatically false

SSBASLD committed:
added a scuffed timer transition

SSBASLD committed:
better timer

SSBASLD committed:
implemented correct state transitions + debouncers

SSBASLD committed:
implemented suggested changes

PotmanNob committed:
autoalign subsystem io abstraction

PotmanNob committed:
w nisala

Commit:
Apply Prettier format

PotmanNob committed:
Merge branch 'Fix-Drive-Configs' into autoAlign-refactor

PotmanNob committed:
minor skill issue

PotmanNob committed:
requested changes

PotmanNob committed:
Merge pull request #31 from FRC-7525/autoAlign-refactor

Auto align refactor

Commit:
Apply Prettier format

GreenTomato5 committed:
it works!!! :)

Commit:
Apply Prettier format

GreenTomato5 committed:
aligned encoder offset stuff

Commit:
Apply Prettier format

GreenTomato5 committed:
good enough - otto

Commit:
Apply Prettier format

GreenTomato5 committed:
odo works now!!!

Commit:
Apply Prettier format

GreenTomato5 committed:
Note vision + PV vendor dep

Commit:
Apply Prettier format

Liquid2112 committed:
Added noteVision into AA with B button

Liquid2112 committed:
Testing note vision code

Liquid2112 committed:
used aa subsystem formatiing

Liquid2112 committed:
Added note testing to NoteVision and not robot

Liquid2112 committed:
Merge pull request #32 from FRC-7525/NoteVision

Note vision

Commit:
Apply Prettier format

Leif031809 committed:
Merge branch 'Fix-Drive-Configs' into beam-breaks

Leif031809 committed:
Merge pull request #28 from SSBASLD/beam-breaks

Beam breaks

Commit:
Apply Prettier format

GreenTomato5 committed:
Re-write un-pushed code

Commit:
Apply Prettier format

GreenTomato5 committed:
Fix Shooting

Commit:
Apply Prettier format

GreenTomato5 committed:
Clean up shooter abstractions

GreenTomato5 committed:
Merge branch 'Fix-Drive-Configs' of https://github.com/FRC-7525/2024-Rewrite into Fix-Drive-Configs

Commit:
Apply Prettier format

GreenTomato5 committed:
Fix note vision

GreenTomato5 committed:
Merge branch 'Fix-Drive-Configs' of https://github.com/FRC-7525/2024-Rewrite into Fix-Drive-Configs

Commit:
Apply Prettier format

GreenTomato5 committed:
Fix intake speed

GreenTomato5 committed:
mid code

Commit:
Apply Prettier format

GreenTomato5 committed:
Make robot unsafe again

GreenTomato5 committed:
remove more saftey

GreenTomato5 committed:
haahahaha

GreenTomato5 committed:
Driver dash + random stuff for testing tmrw

Commit:
Apply Prettier format

GreenTomato5 committed:
Enable camera

GreenTomato5 committed:
Climbers (#37)

* Climbers, lots of magic numbers

* Peak code

* Fix bad grammar

* Nah the code builds, that's lowkey crazy ngl

Commit:
Apply Prettier format

GreenTomato5 committed:
Family - Furious and Fast 18

Commit:
Apply Prettier format

GreenTomato5 committed:
Autos (#34)

* Autos

* Constants and stuff

* Fix insta crash

* Apply Prettier format

* Dubiously functional

* Adds useful functions

* -1,000,000 aura

* Fix L code

---------

Co-authored-by: github-actions <github-actions@github.com>

Commit:
Apply Prettier format

GreenTomato5 committed:
Autos work in sim

GreenTomato5 committed:
Add starting pose incase vision is bum at comp

GreenTomato5 committed:
Speed meter in feet

GreenTomato5 committed:
Remove note vision jibberish (non functional)

Commit:
Apply Prettier format

PotmanNob committed:
score/amp after reaching target (#36)

* score/amp after reaching target

* Fix AA

---------

Co-authored-by: Otto <105565584+GreenTomato5@users.noreply.github.com>
Co-authored-by: Otto <grentomato@gmail.com>

Commit:
Apply Prettier format

GreenTomato5 committed:
Adds April Tag vision (#38)

* Adds AT vision

* Clean up vision

invalid-email-address committed:
Apply Prettier format

GreenTomato5 committed:
Slight fixes

invalid-email-address committed:
Apply Prettier format

GreenTomato5 committed:
+12 aura

invalid-email-address committed:
Apply Prettier format

GreenTomato5 committed:
Climbers working

invalid-email-address committed:
Apply Prettier format

GreenTomato5 committed:
Fix field rel :(

invalid-email-address committed:
Apply Prettier format

GreenTomato5 committed:
Slower than correct speed

invalid-email-address committed:
Apply Prettier format

GreenTomato5 committed:
Plsss fix the robot

invalid-email-address committed:
Apply Prettier format

GreenTomato5 committed:
???

GreenTomato5 committed:
Merge branch 'Fix-Drive-Configs' of https://github.com/FRC-7525/2024-Rewrite into Fix-Drive-Configs

invalid-email-address committed:
Apply Prettier format

GreenTomato5 committed:
If it goes slow ima give up

GreenTomato5 committed:
zesAyt vn cd

invalid-email-address committed:
Apply Prettier format

GreenTomato5 committed:
Merge branches 'Fix-Drive-Configs' and 'Fix-Drive-Configs' of https://github.com/FRC-7525/2024-Rewrite into Fix-Drive-Configs

GreenTomato5 committed:
YAGSL carry 😴

invalid-email-address committed:
Apply Prettier format

GreenTomato5 committed:
Heading correction

GreenTomato5 committed:
disable heading correction before we do testing

GreenTomato5 committed:
Merge branch 'Fix-Drive-Configs' of https://github.com/FRC-7525/2024-Rewrite into Fix-Drive-Configs

invalid-email-address committed:
Apply Prettier format

GreenTomato5 committed:
Clean up wishy washy code

GreenTomato5 committed:
Merge branch 'Fix-Drive-Configs' of https://github.com/FRC-7525/2024-Rewrite into Fix-Drive-Configs

GreenTomato5 committed:
Wow this code is so clean

invalid-email-address committed:
Apply Prettier format

GreenTomato5 committed:
ZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZ

GreenTomato5 committed:
W code

invalid-email-address committed:
Apply Prettier format

GreenTomato5 committed:
??

GreenTomato5 committed:
Fix AA

invalid-email-address committed:
Apply Prettier format

GreenTomato5 committed:
Slighhhht drive PID

SSBASLD committed:
created auto stuff that might work

invalid-email-address committed:
Apply Prettier format

SSBASLD committed:
Revert "created auto stuff that might work"

This reverts commit 6617453916cf1424b9b87147ea40a9ca5ea77db4.

SSBASLD committed:
Merge branch 'Fix-Drive-Configs' of https://github.com/FRC-7525/2024-Rewrite into Fix-Drive-Configs

GreenTomato5 committed:
Tuning

invalid-email-address committed:
Apply Prettier format
